# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1243229719 -7200
#      Mon May 25 07:35:19 2009 +0200
# Node ID 80c299c847967200e320fc15e63c206223844706
# Parent  470af0f93ca9d7947027c42dddac8d1edc220625
don't print undefined symbols in who

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-05-25  Jaroslav Hajek  <highegg@gmail.com>
+
+	* variables.cc (do_who): Only output symbols with a defined value.
+
 2009-05-22  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* toplev.h (main_loop): Tag with OCTINTERP_API.
 	* input.h (octave_read, get_input_from_file, get_input_from_stdin):
 	Ditto.
 	* lex.h (create_buffer, current_buffer, switch_to_buffer,
 	delete_buffer, restore_input_buffer, delete_input_buffer): Ditto.
 
diff --git a/src/variables.cc b/src/variables.cc
--- a/src/variables.cc
+++ b/src/variables.cc
@@ -1500,20 +1500,23 @@ do_who (int argc, const string_vector& a
 	    {
 	      std::list<symbol_table::symbol_record> tmp = global_only
 		? symbol_table::glob_global_variables (pat)
 		: symbol_table::glob_variables (pat);
 
 	      for (std::list<symbol_table::symbol_record>::const_iterator p = tmp.begin ();
 		   p != tmp.end (); p++)
 		{
-		  if (verbose)
-		    symbol_stats.append (*p);
-		  else
-		    symbol_names.push_back (p->name ());
+                  if (p->is_defined ())
+                    {
+                      if (verbose)
+                        symbol_stats.append (*p);
+                      else
+                        symbol_names.push_back (p->name ());
+                    }
 		}
 	    }
 	}
     }
 
   if (return_list)
     {
       if (verbose)
